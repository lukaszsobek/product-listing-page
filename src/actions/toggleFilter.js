import { TOGGLE_FILTER } from '../constants'

const toggleFilter = (filterType, filterValue) => ({
  type: TOGGLE_FILTER,
  filterType,
  filterValue
})

export default toggleFilter
