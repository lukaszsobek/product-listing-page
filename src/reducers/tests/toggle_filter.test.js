import toggleFilter from '../toggle_filter'

describe('toggleFilter reducer', () => {
  const state = {
    activeFilters: {
      aType: ['aFilter']
    }
  }

  it('adds a new filter', () => {
    const action = {
      filterType: 'aType',
      filterValue: 'anotherFilter'
    }

    expect(toggleFilter(state, action)).toMatchSnapshot()
  })

  it('removes existing filter', () => {
    const action = {
      filterType: 'aType',
      filterValue: 'aFilter'
    }

    expect(toggleFilter(state, action)).toMatchSnapshot()
  })
})
