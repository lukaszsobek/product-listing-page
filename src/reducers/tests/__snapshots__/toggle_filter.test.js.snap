// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toggleFilter reducer adds a new filter 1`] = `
Object {
  "activeFilters": Object {
    "aType": Array [
      "aFilter",
      "anotherFilter",
    ],
  },
}
`;

exports[`toggleFilter reducer removes existing filter 1`] = `
Object {
  "activeFilters": Object {
    "aType": Array [],
  },
}
`;
